ops/model.py

```diff
@@ -985,6 +1006,52 @@ def location(self):
         return self._location
 
 
+class Container:
+    """Represents a named (Kubernetes) container in an application.
+
+    Attributes:
+        name: The name of the container from metadata.yaml (eg, 'postgres').
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+        socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
+        self._pebble = pebble.API(socket_path=socket_path)
+
+    @property
+    def pebble(self) -> 'pebble.API':
+        """Return the low-level Pebble API instance for this container."""
+        return self._pebble
+
+    def autostart(self):
+        """Autostart all default services."""
+        self._pebble.autostart_services()
+
+    def start(self, *service_names: typing.List[str]):
+        """Start given service(s) by name."""
+        self._pebble.start_services(service_names)
+
+    def stop(self, *service_names: typing.List[str]):
+        """Stop given service(s) by name."""
+        self._pebble.stop_services(service_names)
+
+    def add_layer(self, config: typing.Union[str, typing.Dict, 'pebble.Config']):
+        """Dynamically add a service (Pebble) configuration layer.
```

Comment:
After sleeping on it over the weekend, I really think we want a different name than "config" here, because that already has a very specific meaning in juju.  It doesn't necessarily have to be "setup", though. We just need something nice that isn't config.

Reply:
Yep, I've changed this to `Setup` to match Pebble's terminology and avoid confusion with Juju config.